WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/weapon/reload/ammo/Magazine.java

```diff
@@ -0,0 +1,81 @@
+package me.deecaad.weaponmechanics.weapon.reload.ammo;
+
+import me.deecaad.weaponmechanics.WeaponMechanics;
+import me.deecaad.weaponmechanics.WeaponMechanicsAPI;
+import org.bukkit.ChatColor;
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.Damageable;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.bukkit.persistence.PersistentDataType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class Magazine {
+    private final ItemAmmo itemAmmo;
+    private final ItemStack baseItem;
+
+    public Magazine(ItemAmmo itemAmmo, ItemStack baseItem) {
+        this.itemAmmo = itemAmmo;
+        this.baseItem = baseItem;
+    }
+
+    public static int getAmmoFromItem(ItemStack item) {
+        ItemMeta itemMeta = item.getItemMeta();
+        PersistentDataContainer dataContainer = itemMeta.getPersistentDataContainer();
+        return dataContainer.getOrDefault(new NamespacedKey(WeaponMechanics.getPlugin(), "ammo"), PersistentDataType.INTEGER, 0);
+    }
+
+    public static int getAmmo(ItemStack weapon, ItemStack item) {
+        return Magazine.getCapacity(item) <= 0 ? Magazine.getCapacity(WeaponMechanicsAPI.getWeaponTitle(weapon)) : Magazine.getAmmoFromItem(item);
+    }
+
+    public static int getCapacity(String weaponTitle) {
+        return WeaponMechanics.getConfigurations().getInt(weaponTitle + ".Reload.Magazine_Size");
```

Comment:
serialized objects should not pull values from config... And why is this static?? and why is this acting on a weapon title and not a magazine??? Overall really bad code.